test/jdk/java/nio/channels/DatagramChannel/PromiscuousIPv6.java

```diff
@@ -196,17 +203,14 @@ static void test(ProtocolFamily family,
 
     public static void main(String[] args) throws IOException {
 
-        String os = System.getProperty("os.name");
+        boolean hasIPV6MulticastAll;
 
-        if (!os.equals("Linux")) {
-            throw new SkippedException("This test should be run only on Linux");
+        if (Platform.isWindows()) {
+            throw new SkippedException("This test should not be run on Windows");
         } else {
-            String osVersion = System.getProperty("os.version");
-            String prefix = "3.10.0";
-            if (osVersion.startsWith(prefix)) {
-                throw new SkippedException(
-                        String.format("The behavior under test is known NOT to work on '%s' kernels", prefix));
-            }
+            int major = Platform.getOsVersionMajor();
+            int minor = Platform.getOsVersionMinor();
+            hasIPV6MulticastAll = Platform.isOSX() || (major > 4) || ((major == 4 && minor >= 20));
```

Comment:
The version check should be Linux only.

Reply:
It runs the additional tests if it is MacOSX or if Linux and kernel > 4.20.

Reply:
It looks like it's good the version check on AIX too. I think you'll need to include a check for isLinux to avoid a follow-up bug from the maintainers of that port.

Reply:
Oh, I see there is a missing pair of braces. I will fix that

Reply:
Ok then. I'll add explicit tests for Linux and Windows, and the AIX folks can add that if/when they verify it works.